src/python/pants/engine/native.py

```diff
@@ -517,6 +538,36 @@ class Function(NamedTuple):
   key: Key
 
 
+class EngineTypes(NamedTuple):
+  """Python types that need to be passed to the engine. EngineTypes corresponds field-by-field to
+  the Types struct defined in `src/rust/engine/src/types.rs` (field definition order matters, not just name!)."""
```

Comment:
In general, docstrings are typically formatted with a single initial sentence, and further text after a newline. This isn't necessary to follow in 100% of cases, but I think that formatting could be effective here. That would look like:
```python
class EngineTypes(NamedTuple):
  """Python types that need to be passed to the engine.

  NB: EngineTypes corresponds field-by-field to the Types struct defined in `src/rust/engine/src/types.rs`
  (field definition order matters, not just name!).
  """
```